mlflow/server/js/src/components/artifact-view-components/ShowArtifactHtmlView.js

```diff
@@ -0,0 +1,88 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { getSrc } from './ShowArtifactPage';
+import './ShowArtifactHtmlView.css';
+import { getRequestHeaders } from '../../setupAjaxHeaders';
+import Iframe from 'react-iframe'
+
+class ShowArtifactHtmlView extends Component {
+  constructor(props) {
+    super(props);
+    this.fetchArtifacts = this.fetchArtifacts.bind(this);
+  }
+
+  static propTypes = {
+    runUuid: PropTypes.string.isRequired,
+    path: PropTypes.string.isRequired,
+  };
+
+  state = {
+    loading: true,
+    error: undefined,
+    html: undefined,
+  };
+
+  componentWillMount() {
+    this.fetchArtifacts();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.path !== prevProps.path || this.props.runUuid !== prevProps.runUuid) {
+      this.fetchArtifacts();
+    }
+  }
+
+  render() {
+    if (this.state.loading) {
+      return (
+        <div>
+          Loading...
+        </div>
+      );
+    }
+    if (this.state.error) {
+      return (
```

Comment:
Could we also log the error to help users debug/give us diagnostic information in the event of failures? e.g.
```logger.error("Unable to load HTML artifact, got error " + this.state.error")```?

Reply:
Do you expect me to print the error on the web console?
`console.error("Unable to load HTML artifact, got error " + this.state.error);`

Reply:
Yep, let's add that line before we `return` here, e.g.

```suggestion
      console.error("Unable to load HTML artifact, got error " + this.state.error);
      return (
```

assuming that `this.state.error` is in fact a string (would be great if you could confirm this) - thanks!

Reply:
I have copy the fetchArtifacts() function from the ShowArtifactTextView.js file. I currently can't see where an error is being set.

Reply:
I add a catch() method to set the error in a error case